robottelo/common/helpers.py

```diff
@@ -117,6 +123,27 @@ def get_server_cert_rpm_url():
         get_server_pub_url(), 'katello-ca-consumer-latest.noarch.rpm')
 
 
+def get_host_info(hostname=None):
+    """Get remote host's distribution information
+
+    :param str hostname: Hostname or IP address of the remote host. If ``None``
+        the hostname will be get from ``main.server.hostname`` config.
+
+    """
+    result = ssh.command('cat /etc/redhat-release', hostname)
+    if result.return_code != 0:
+        raise HostInfoError('/etc/redhat-release not found')
+    match = re.match(
+        r'(?P<distro>.+) release (?P<major>\d+)(.(?P<minor>\d+))?',
+        result.stdout[0],
+    )
+    if match is None:
+        raise HostInfoError(
+            u'Not able to parse release string "{0}"'.format(result.stdout[0]))
+    groups = match.groupdict()
+    return (groups['distro'], groups['major'], groups['minor'])
```

Comment:
Your code has a regex that looks for integer-like numbers. Given that you explicitly search for integer-like numbers and raise an exception otherwise, I'd say "yes, convert the found version numbers to integers." Doing so allows for more straightforward version comparisons in simple cases. And in more complex cases where users create a `packaging.version.Version` object (or call `packaing.version.parse`), users will be able to work with either strings or integers.

/me wonders if Satellite has a policy for the form of their version numbers.